include/solvers/memory_solution_history.h

```diff
@@ -22,6 +22,7 @@
 
 // Local includes
 #include "libmesh/numeric_vector.h"
+#include "libmesh/diff_system.h"
```

Comment:
This should be in the .C file, since only the .C file needs it; no need to force every other memory_solution_history.h-using file to include it too.